libs/sarif/src/schema/tool.rs

```diff
@@ -149,6 +149,14 @@ impl Tool {
 pub struct ToolComponent {
```

Comment:
`3.19.2`: "At least one of version (§3.19.13) and semanticVersion (§3.19.12) SHOULD be present" this invariant is not upheld in the constructor (or by having publicly accessible fields)

Reply:
This is wrong, I (once again), have mistaken `SHOULD` for `SHALL` (which while being very similar have a totally different meaning). So we do not need to ensure that at least one of them exists. My bad! I am fine with the current version!